Frameworks/Foundation/NSIndexSet.mm

```diff
@@ -483,18 +484,78 @@ - (NSUInteger)indexGreaterThanOrEqualToIndex:(NSUInteger)index {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (NSUInteger)getIndexes:(NSUInteger*)indexBuffer maxCount:(NSUInteger)bufferSize inIndexRange:(NSRangePointer)indexRange {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (indexBuffer == nullptr || bufferSize == 0) {
+        return 0;
+    }
+
+    __block NSUInteger index = 0;
+    NSRange enumerationRange = indexRange ? *indexRange : sc_unlimitedRange;
+    [self enumerateRangesInRange:enumerationRange
+                         options:0
+                      usingBlock:^(NSRange range, BOOL* stop) {
+                          NSUInteger j = range.location;
+                          for (; j < range.location + range.length && index < bufferSize; ++j, ++index) {
+                              indexBuffer[index] = j;
+                          }
+
+                          if (indexRange) {
+                              *indexRange = { j, range.location + range.length - j };
+                          }
+
+                          // Should never have case where index > bufferSize but better safe than sorry
+                          if (index >= bufferSize) {
+                              *stop = YES;
+                          }
+                      }];
+
+    return index;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (void)enumerateIndexesInRange:(NSRange)range options:(NSEnumerationOptions)opts usingBlock:(void (^)(NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+- (void)enumerateIndexesInRange:(NSRange)range options:(NSEnumerationOptions)options usingBlock:(void (^)(NSUInteger, BOOL*))block {
+    dispatch_queue_t queue;
+    dispatch_group_t group;
+
+    // Initialize dispatch queue for concurrent enumeration
+    if (options & NSEnumerationConcurrent) {
+        queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+        group = dispatch_group_create();
+    }
+
+    BOOL reverse = options & NSEnumerationReverse;
+    NSUInteger count = [static_cast<NSMutableIndexSet*>(self) _count];
+    NSRangePointer ranges = [static_cast<NSMutableIndexSet*>(self) _allRanges];
+    NSUInteger start = (reverse) ? count : 1;
+    NSUInteger end = (reverse) ? 0 : count + 1;
+    NSInteger step = (reverse) ? -1 : 1;
+    __block BOOL stop = NO;
+    for (NSUInteger i = start; i != end && stop == NO; i += step) {
+        NSRange intersection = NSIntersectionRange(range, ranges[i - 1]);
+        if (intersection.length != 0) {
+            NSUInteger rangeStart = reverse ? intersection.location + intersection.length : intersection.location + 1;
```

Comment:
you're right, this DOES seem like it should be able to be lifted out/commonized.